doc/architecture/decisions/0008-data-manipulation-for-views-should-happen-in-util-files.md

```diff
@@ -0,0 +1,23 @@
+# 7. Data manipulation for views should happen in util files
+
+Date: 2023-02-14
+
+## Status
+
+Proposed
+
+## Context
+
+Nunjucks is limited as to how much manipulation it allows and therefore some kinds of neccessary manipulation can only happen in Node.
+There isn't a clear precedent in HMPPS as to where such manipulation should happen.
+We initially did such manipulation in the services but as the app has evolved we now believe that
+
+## Decision
+
+Manipulation of data for views that isn't possible in Nunjucks should happen in individual utils which are imported into the Nunjucks templates.
```

Comment:
Would be good if we could talk about _how_ we import these into the template - we're starting to lean towards importing whole modules. Also, as a rule, we're importing them as globals, rather than filters

Reply:
Yeah that's a good point

Reply:
I'm not clear on why we're importing them as functions over filters - it's not something I've done intentionally. Do you have any reasoning? 
I think importing whole modules makes sense as there is less boilerplate when adding a util function but I've not really thought about filters vs globals

Reply:
Filters can only accept one argument eg:

```
{{ variable | doSomething }}
```

so I've tended to steer clear of them, also, the syntax is weird